src/Microsoft.VisualStudio.Threading/AsyncReaderWriterLock.cs

```diff
@@ -1718,10 +1818,35 @@ private void PendAwaiter(Awaiter awaiter)
                 {
                     Queue<Awaiter>? queue = this.GetLockQueue(awaiter.Kind);
                     queue.Enqueue(awaiter);
+
+                    if (awaiter.Kind == LockKind.Write)
+                    {
+                        this.EnablePendingWriterLockDeadLockWatching();
+                    }
                 }
             }
         }
 
+        private void EnablePendingWriterLockDeadLockWatching()
```

Comment:
Maybe call this `StartPendingWriterDeadlockTimerIfNecessary()` to reflect that conditions are applied before it decides to actually start the timer.

Reply:
yeah, i made that change.